docs/source/conf.py

```diff
@@ -441,15 +453,289 @@ def check_timeout_and_api_kwargs_presence(obj: object) -> int:
+        """
+        return str(eval(short_class_name))
+
+    # First, generate a mapping of class names to the attributes in other classes that correspond
+    # to instances of a given class
+    # i.e. {"telegram.Sticker": ["telegram.Message.sticker", ...]}
+    attrs_for_class_name = defaultdict(list)
+    admonition_for_class_name = {}
+
+    # The colon after the closing parenthesis is important because it makes sure
+    # that the class is mentioned in the attribute description, not in free text.
+    # So far there seem to be no attribute docstrings in which the list of classes
+    # takes up more than one line.
+    attr_docstr_pattern = re.compile(r"^\s*(?P<attr_name>[a-z_]+)\s?\(.*:class:`.+`.*\):\s.*$")
+    prop_docst_pattern = re.compile(r":class:`.+`.*:")
+
+    single_class_name_pattern = re.compile(r":class:`~?(?P<class_name>[\w.]*)`")
+
+    for _, inspected_class in inspect.getmembers(telegram, inspect.isclass) + inspect.getmembers(
+        telegram.ext, inspect.isclass
+    ):
+        # We need to make "<class 'telegram._files.sticker.StickerSet'>" into "telegram.StickerSet"
+        # because that's the way the classes are mentioned in docstrings...
+        name_of_inspected_class_in_docstr = (
+            str(inspected_class).removeprefix("<class '").removesuffix("'>")
+        )
+        name_of_inspected_class_in_docstr = ".".join(
+            n
+            for n in name_of_inspected_class_in_docstr.split(".")
+            if n in ("telegram", "ext") or not n.islower()  # removing things like "_files.sticker"
+        )
+
+        # Parsing part of the docstring with attributes (parsing of properties follows later)
+        docstring_lines = inspect.getdoc(inspected_class).splitlines()
+        lines_with_attrs = []
+        for idx, line in enumerate(docstring_lines):
+            if line.strip() == "Attributes:":
+                lines_with_attrs = docstring_lines[idx + 1 :]
+
+        for line in lines_with_attrs:
+            line_match = attr_docstr_pattern.match(line)
+            if not line_match:
+                continue
+
+            target_attr = line_match.group("attr_name")
+            # a typing description of one attribute can contain multiple classes
+            for match in single_class_name_pattern.finditer(line):
+                name_of_class_in_attr = match.group("class_name")
+                if "telegram" not in name_of_class_in_attr:
+                    continue
+
+                # Writing to dictionary: matching the class found in the docstring to the
+                # attribute of the class being inspected.
+                # The class in the attribute docstring is the key, and the attribute of the
+                # class currently being inspected is the value.
+                attrs_for_class_name[build_full_class_name(name_of_class_in_attr)].append(
+                    f":attr:`{name_of_inspected_class_in_docstr}.{target_attr}`"
```

Comment:
what happens if the entire name is linked, i.e. `` :attr:`telegram._files.sticker.StickerSet` ``, does it link in the docs? Does it show as that or as `telegram.StickerSet`?

Reply:
The method for resolving the full class name (in this older version it was a function called `build_full_class_name()`) first tries to evaluate the entire name:
```py 
for option in (name, f"telegram.{name}", f"telegram.ext.{name}"):
    try:
        return str(eval(option))
```
So it does work if the entire name is linked.

As for how it shows - I guess it is Sphinx's behavior: all full class names that I return from the aforementioned method look like this:
```
<class 'telegram._keyboardbuttonpolltype.KeyboardButtonPollType'>
```
And when I generate the links, Sphinx shows then in rendered docs as `telegram.KeyboardButtonPollType`